src/compiler/checker.ts

```diff
@@ -1526,17 +1534,53 @@ module ts {
                     }
                 }
 
+                function writeSymbolTypeReference(symbol: Symbol, typeArguments: Type[], pos: number, end: number) {
+                    if (!isReservedMemberName(symbol.name)) {

```

Comment:
This happens with unnamed generic function expressions, generic arrow functions, and unnamed generic class expressions.